### PR TITLE
Alerting: Reduce set of fields that could trigger alert state change

### LINKED ISSUE #83250: Reduce set of fields which changes in alert rule that cause the reset of state.
It's well known that when a user updates a rule, the alert instances produced by that rule get reset, and active alerts are resolved.

Historically, the reason we reset the alert state and resolve notifications every time the rule is changed is that the new rule could produce a different state, and states of the old version of the rule could be orphaned and stick around for some time. That could confuse users and cause multiple false-positive notifications. 

However, the resetting of the state makes sometimes unnecessary noise in state history, and notifications. For example:
- https://github.com/grafana/grafana/issues/64256
- https://github.com/grafana/grafana/issues/79235
- https://github.com/grafana/grafana/pull/66531#issuecomment-1952777882
- + some escalations from customers.

Since PR https://github.com/grafana/grafana/pull/66531 that upgraded the state manager to be more precise in decisions about whether rule change should reset the state, we got more control over changes in which fields should cause the state to be reset, and could reduce unnecessary resets.

However, there is the possibility to reduce it even more if we exclude some fields from the fingerprint. I have identified the list of these fields 
https://github.com/grafana/grafana/blob/32fde6dba4be14fc90d58b07d97d5b3585b50d86/pkg/services/ngalert/schedule/registry.go#L312-L326

Let's remove them from the fingerprint and narrow down the situations when rule change causes the state reset.

### PR DESCRIPTION
**Pull request by benoittgt**
**What is this feature?**

We want to avoid too much change of alert state based on change on alert's fields. For that we ignore some fields from the diff.

This is a rather cautious first pull request, ~~but I would love to make another pull request for `IntervalSeconds`~~. We could even go further as [pointed](https://github.com/grafana/grafana/issues/83250) by @yuri-tceretian but I can do in another PR.

**Why do we need this feature?**

There is great issue by @yuri-tceretian  that detail all fo this https://github.com/grafana/grafana/issues/83250

**Who is this feature for?**

Everyone who use alerting and updates.

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/grafana/issues/83250

cc @yuri-tceretian

### CODE REVIEW

File: pkg/services/ngalert/schedule/registry.go [hunk 1]
```diff
@@ -326,7 +308,6 @@ func (r ruleWithFolder) Fingerprint() fingerprint {
 	writeString(rule.Title)
 	writeString(rule.NamespaceUID)
 	writeString(r.folderTitle)
-	writeLabels(rule.Labels)
```
Comment (grobinson-grafana) [inline pkg/services/ngalert/schedule/registry.go:329]: Looking at Yuri's comments, we cannot remove labels as it is fundamental to the fingerprint.

File: pkg/services/ngalert/schedule/registry.go [hunk 2]
```diff
@@ -347,7 +328,6 @@ func (r ruleWithFolder) Fingerprint() fingerprint {
 	writeInt(rule.OrgID)
 	writeInt(rule.IntervalSeconds)
 	writeInt(int64(rule.For))
-	writeLabels(rule.Annotations)
 	if rule.DashboardUID != nil {
 		writeString(*rule.DashboardUID)
 	}
```
Comment (yuri-tceretian) [inline pkg/services/ngalert/schedule/registry.go:333]: Only fields below //TODO are up for removal. Other fields cannot be removed.

File: pkg/services/ngalert/schedule/registry.go (hunk 1 above)
Comment (benoittgt) [reply pkg/services/ngalert/schedule/registry.go:329]: You are correct thanks.

**Review by yuri-tceretian — APPROVED** [review]
LGTM. Thanks for the contribution! I really appreciate it!
We also can delete DashboardUID and PanelID because they are the product of annotations. Removal of ID could help us handle situations when a rule is deleted and a new one with the same UID is created. No need to address it here. We can open another PR.

File: pkg/services/ngalert/schedule/registry_test.go [hunk 3]
```diff
@@ -369,10 +369,12 @@ func TestRuleWithFolderFingerprint(t *testing.T) {
 		f2 := ruleWithFolder{rule: rule, folderTitle: uuid.NewString()}.Fingerprint()
 		require.NotEqual(t, f, f2)
 	})
-	t.Run("Version and Updated should be excluded from fingerprint", func(t *testing.T) {
+	t.Run("Version, Updated, IntervalSeconds and Annotations should be excluded from fingerprint", func(t *testing.T) {
 		cp := models.CopyRule(rule)
 		cp.Version++
 		cp.Updated = cp.Updated.Add(1 * time.Second)
+		cp.IntervalSeconds++
+		cp.Annotations["test"] = "test"
```
Comment (yuri-tceretian) [inline pkg/services/ngalert/schedule/registry_test.go:377]: The failure is here. Please cerate a new annotation map.

**Comment by yuri-tceretian** [comment]
Unfortunately, I forgot to merge it before Grafana 11 cutoff. This will be included only in Grafana 11.1

**Comment by benoittgt** [comment]
@yuri-tceretian no problem. Thanks for the merge. Do you know when 11.1 will be released?

**Comment by ashharrison90** [comment]
@benoittgt this actually did make it within the cutoff, so will be released with grafana 11 👍

**Pull request merged**